platform/security/platform-security-core-api/src/main/java/ddf/security/samlp/MetadataConfigurationParser.java

```diff
@@ -238,7 +240,20 @@ private EntityDescriptor readEntityDescriptor(Reader reader) {
       throw new IllegalArgumentException(
           "Unable to convert EntityDescriptor document to XMLObject.");
     }
+    EntityDescriptor root = (EntityDescriptor) entityXmlObj;
+    validateMetadata(root);
+    return root;
+  }
 
-    return (EntityDescriptor) entityXmlObj;
+  private void validateMetadata(EntityDescriptor root) {
+    boolean valid =
+        (root.getCacheDuration() != null && root.getCacheDuration() > 0)
+            || DateTime.now().isBefore(root.getValidUntil());
+    if (!valid) {
+      LOGGER.info(
+          "IDP metadata must either have cache duration or future valid-until date."
+              + " Setting cache duration to 24 hours.");
+      root.setCacheDuration(ONE_DAY_MILLISECONDS);
```

Comment:
I am not very familiar with this part of the code but is it ok to just set the catch duration like this? Should we throw an exception instead?